programs/uxd/src/instructions/rebalance_alloyx_vault_depository.rs

```diff
@@ -0,0 +1,773 @@
+use crate::state::alloyx_vault_depository::AlloyxVaultDepository;
+use crate::state::credix_lp_depository::CredixLpDepository;
+use crate::state::identity_depository::IdentityDepository;
+use crate::state::mercurial_vault_depository::MercurialVaultDepository;
+use crate::utils::calculate_router_depositories_target_redeemable_amount;
+use crate::utils::checked_add;
+use crate::utils::checked_sub;
+use crate::utils::compute_decrease;
+use crate::utils::compute_increase;
+use crate::utils::compute_shares_amount_for_value_floor;
+use crate::utils::compute_value_for_shares_amount_floor;
+use crate::utils::is_within_range_inclusive;
+use crate::validate_is_program_frozen;
+use crate::Controller;
+use crate::ALLOYX_VAULT_DEPOSITORY_NAMESPACE;
+use crate::CONTROLLER_NAMESPACE;
+use crate::IDENTITY_DEPOSITORY_NAMESPACE;
+use anchor_lang::prelude::*;
+use anchor_spl::associated_token::AssociatedToken;
+use anchor_spl::token;
+use anchor_spl::token::Mint;
+use anchor_spl::token::Token;
+use anchor_spl::token::TokenAccount;
+use anchor_spl::token::Transfer;
+
+#[derive(Accounts)]
+pub struct RebalanceAlloyxVaultDepository<'info> {
+    /// #1
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    /// #2 - This account will be responsible for paying the precision loss
+    #[account(
+        mut,
+        token::authority = payer,
+        token::mint = collateral_mint,
+    )]
+    pub payer_collateral: Box<Account<'info, TokenAccount>>,
+
+    /// #3
+    #[account(
+        mut,
+        seeds = [CONTROLLER_NAMESPACE],
+        bump = controller.load()?.bump,
+        constraint = controller.load()?.identity_depository == identity_depository.key() @UxdError::InvalidDepository,
+        constraint = controller.load()?.mercurial_vault_depository == mercurial_vault_depository.key() @UxdError::InvalidDepository,
+        constraint = controller.load()?.credix_lp_depository == credix_lp_depository.key() @UxdError::InvalidDepository,
+        constraint = controller.load()?.alloyx_vault_depository == alloyx_vault_depository.key() @UxdError::InvalidDepository,
+    )]
+    pub controller: AccountLoader<'info, Controller>,
+
+    /// #4
+    pub collateral_mint: Box<Account<'info, Mint>>,
+
+    /// #5
+    #[account(
+        mut,
+        has_one = collateral_mint @UxdError::InvalidCollateralMint,
+        constraint = identity_depository.load()?.collateral_vault == identity_depository_collateral.key() @UxdError::InvalidDepositoryCollateral,
+    )]
+    pub identity_depository: AccountLoader<'info, IdentityDepository>,
+
+    /// #6
+    #[account(mut)]
+    pub identity_depository_collateral: Box<Account<'info, TokenAccount>>,
+
+    /// #7
+    #[account(
+        mut,
+        has_one = controller @UxdError::InvalidController,
+        has_one = collateral_mint @UxdError::InvalidCollateralMint,
+    )]
+    pub mercurial_vault_depository: AccountLoader<'info, MercurialVaultDepository>,
+
+    /// #8
+    #[account(
+        mut,
```

Comment:
I think we can remove the `mut` for the depository

Reply:
nice catch! done!